app/controllers/application_controller.rb

```diff
@@ -2,4 +2,32 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  before_action :logged_in?, except: [:index, :show]
+
+  helper_method :current_user, :author?, :asker?
+
+
+  def current_user
+  	User.find(session[:user_id]) if session[:user_id]
+  end
+
+  def author?(answer)
+    current_user && (current_user.id == answer.user_id)
```

Comment:
I'd put this on the model itself. So on answer add
```
def author?(a_user)
  self.user == a_user
end
```

ActiveRecord does sensible equals methods and will compare the id